src/bio/smd/smd_pool.c

```diff
@@ -408,3 +438,35 @@ smd_pool_replace_blobs_locked(struct smd_pool_info *info, int tgt_cnt,
 	}
 	return rc;
 }
+
+int
+smd_pool_mark_ready(uuid_t pool_id)
+{
+	struct smd_pool	pool;
+	struct d_uuid	id;
+	int		rc;
+
+	uuid_copy(id.uuid, pool_id);
+
+	smd_db_lock();
+	rc = smd_db_fetch(TABLE_POOLS[SMD_DEV_TYPE_META], &id, sizeof(id), &pool, sizeof(pool));
+	if (rc == -DER_NONEXIST) {
+		D_GOTO(out, rc = 0);
+	} else if (rc) {
+		D_ERROR("Failed to fetch smd entry of the meta blob for "DF_UUID". "DF_RC"\n",
+			DP_UUID(&id.uuid), DP_RC(rc));
+		goto out;
+	}
+
+	pool.sp_flags &= ~SMD_POOL_IN_CREATION;
```

Comment:
Minor: Instead of clearing 'in_create" flag, if you set a 'ready' flag here, the code for marking 'in_create' in pool_add_tgt() could be avoided.

Reply:
I coded the above way to handle scenario where in if one upgrades the daos bits with smd from previous bits, it continue to work fine. Else the new server would assume that the pools are newborn.